cmd/server/controllers/ProductController.go

```diff
@@ -0,0 +1,129 @@
+package controllers
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/vinigracindo/mercado-fresco-stranger-strings/internal/domains/product"
+	"net/http"
+	"strconv"
+)
+
+type ProductController struct {
+	service product.Service
+}
+
+func CreateProductController(prodService product.Service) *ProductController {
+	return &(ProductController{service: prodService})
+}
+
+func (c *ProductController) GetAll() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		products, err := c.service.GetAll()
+		if err != nil {
+			ctx.JSON(http.StatusNotFound, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+		ctx.JSON(http.StatusOK, products)
+	}
+}
+
+func (c *ProductController) GetById() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		id, err := strconv.ParseInt(ctx.Param("id"), 10, 64)
+		if err != nil {
+			ctx.JSON(http.StatusBadRequest, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+
+		product, err := c.service.GetById(id)
+		if err != nil {
+			ctx.JSON(http.StatusNotFound, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+		ctx.JSON(http.StatusOK, *product)
+	}
+}
+
+func (c *ProductController) Create() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		var req request
+		if err := ctx.ShouldBindJSON(&req); err != nil {
+			ctx.JSON(http.StatusUnprocessableEntity,
+				gin.H{
+					"message": "Input inválido. Verifique os dados digitados",
+				})
+			return
+		}
+		newProduct, err := c.service.Create(req.ProductCode, req.Description, req.Width, req.Height, req.Length, req.NetWeight,
+			req.ExpirationRate, req.RecommendedFreezingTemperature, req.FreezingRate, req.ProductTypeId, req.SellerId)
+
+		if err != nil {
+			ctx.JSON(http.StatusNotFound, gin.H{
```

Comment:
Como você  precisa verificar se já existe o id no banco, se existir, o mais correto é retornar um 409 (conflict) ao invés  de 404 (not found).